added _persistEEPROM

bwjohns4 opened the pull request:
This allows users who are already using EEPROM to continue in an intuitive way and not have this library close the EEPROM by calling EEPROM.end() within the library. Now in the constructor if you set persistent = true then this library will commit() instead of .end().

krzychb reviewed (changes requested):
Thank you for noticing #14 and preparing the updates @bwjohns4!
Could you please add [braces](https://stackoverflow.com/questions/12193170/whats-the-purpose-of-using-braces-i-e-for-a-single-line-if-or-loop) to the newly added if statements?

krzychb commented on src/EspSaveCrash.cpp:
```diff
@@ -200,7 +203,8 @@ void EspSaveCrash::print(Print& outputDev)
   int16_t writeFrom;
   EEPROM.get(_offset + SAVE_CRASH_WRITE_FROM, writeFrom);
-  EEPROM.end();
+  if(!_persistEEPROM) EEPROM.end();
+  else EEPROM.commit();
```
Braces are missing.

krzychb commented on src/EspSaveCrash.cpp:
```diff
@@ -262,7 +266,8 @@ int EspSaveCrash::count()
   EEPROM.begin(_offset + _size);
   int crashCounter = EEPROM.read(_offset + SAVE_CRASH_COUNTER);
-  EEPROM.end();
+  if(!_persistEEPROM) EEPROM.end();
+  else EEPROM.commit();
```
Braces are missing.

krzychb commented on src/EspSaveCrash.h:
```diff
@@ -96,6 +96,7 @@ class EspSaveCrash
+    static bool _persistEEPROM;
   private:
     // none
 };
```
Please squash commit after doing the updates.

krzychb commented:
@bwjohns4,

Thank you for taking the time to submit this PR.

The updates look very good!

Please consider updating [Library Configuration](https://github.com/krzychb/EspSaveCrash/blob/master/examples.md#library-configuration) section of the documentation with the newly added parameter.  I think users may miss the extra functionality you provided if it is not documented.

bwjohns4 commented:
I added the braces and updated the ReadMe. Please let me know if anything else is needed. I'm not sure how to squash the commits since I've already pushed to the remote. Are you able to do that when you merge this in?

krzychb commented:
@bwjohns4, thank you for the quick updates!
